testing/python/fixture.py

```diff
@@ -4046,3 +4046,287 @@ def test_2(fix):
         )
         reprec = testdir.inline_run()
         reprec.assertoutcome(passed=2)
+
+
+@pytest.mark.issue("https://github.com/pytest-dev/pytest/issues/1552")
+class TestFixtureParametrizationAndAutouseUsages(object):
```

Comment:
Great job here as well! 👍

Reply:
A tip: you can add these codes as plain test files in `testing/example_scripts`; this makes it more easy to debug them.

Reply:
oh haha thanks! yeah, debugging them was kinda odd for me. I've never used tox before.